plugins/scaffolder-backend/src/scaffolder/tasks/TaskWorker.test.ts

```diff
@@ -172,6 +173,73 @@ describe('TaskWorker', () => {
     expect((event?.body?.output as JsonObject).result).toBe('winning');
   });
 
+  it('should execute steps conditionally', async () => {
+    const broker = new StorageTaskBroker(storage, logger);
+    const taskWorker = new TaskWorker({
+      logger,
+      workingDirectory: os.tmpdir(),
+      actionRegistry,
+      taskBroker: broker,
+    });
+
+    const { taskId } = await broker.dispatch({
+      steps: [
+        { id: 'test', name: 'test', action: 'test-action' },
+        {
+          id: 'conditional',
+          name: 'conditional',
+          action: 'test-action',
+          if: '"{{ steps.test.output.testOutput }}"',
```

Comment:
This is completely correct. It will always to evaluate to true, as long as it's not an empty string. I think this is a mistake I made in the test, which still leads to a positive result.

`"false"` is treated as a string by this implementation and `false` as a boolean. You can add quotes if you still want it to be a string. In general I think one should avoid quoting here, as it isn't save if the quoted string contains quotes (think about escaping). The build in `json` helper could be used to quote stuff safely. 

The problem is, that the `if` condition works different from the other places where handlebar templates are evaluated. In inputs and outputs, templates are only parsed conditionally as JSON if they start with `"`, "{", "[". That behavior makes sense in these places, but not parsing booleans for conditions would be strange. Instead of parsing them as JSON we could also treat the boolean strings special (e.g. `false` if `input === 'false'`), but then there are also numbers like `0` that have a special meaning.

But I'm open to suggestions

I fixed the test by removing the quotes. The next test, which tests a negative condition, is already without the quotes.